configure.ac

```diff
@@ -113,6 +113,15 @@ AC_CHECK_FUNC([pthread_spin_init],
 	[have_spinlock=1],
 	[have_spinlock=0])
 
+dnl shm_open not used in the common code on os-x
+
+if test "$macos" -eq "0"; then
```

Comment:
Another nit: if you use `-eq`, you are guaranteeing that `test` is testing integer values, so you shouldn't need quotes.  Meaning: you're guaranteeing that `$macos` is *always* 0 or 1 (i.e., it's never empty).  Put differently: if you need the quotes here, then `-eq` is not always going to do what you expect.